client/web/compose/src/components/PageBlocks/CalendarBase.vue

```diff
@@ -355,6 +355,15 @@ export default {
 
       const route = { name: 'page.record', params: { recordID, pageID: page.pageID } }
 
+      if (this.inModal) {
+        this.$root.$emit('show-record-modal', {
+          recordID,
+          recordPageID: page.pageID,
+        })
+
+        return
+      }
```

Comment:
Just a small thing, if you could chain this with the if below instead of returning after.
Since its one of the cases so it makes sense its if/else-if

Same for all other occurrences of this